test/js_test.go

```diff
@@ -630,15 +635,22 @@ func TestJetStreamAckPending_Pull(t *testing.T) {
 
 	// 3 messages delivered 5 times.
 	expected := 15
```

Comment:
Currently asking for 15 msgs with MaxkAckPending==3, means waiting for 5 redeliveries.  After including this change from the server this would work: https://github.com/nats-io/nats-server/pull/1958/files

```
	_, err = sub.Fetch(15)
	if err == nil {
		t.Fatal("Unexpected success")
	}

	// 3 messages re-delivered 5 times.
	expected := 15
	pending := 0
	msgs := make([]*nats.Msg, 0)
	timeout := time.Now().Add(2 * time.Second)
	for time.Now().Before(timeout) {
		ms, err := sub.Fetch(3)
		if err != nil || (ms != nil && len(ms) == 0) {
			continue
		}

		msgs = append(msgs, ms...)
		if len(msgs) >= expected {
			break
		}
		time.Sleep(10 * time.Millisecond)
	}
```